examples/mother/lib.rs

```diff
@@ -0,0 +1,219 @@
+        /// to take snapshots of on arbitrary length (in blocks)
+        EndingPeriod(BlockNumber),
+        /// Candle was blown
+        Ended,
+        /// We have completed the bidding process and are waiting for the Random Function to return some acceptable
+        /// randomness to select the winner. The number represents how many blocks we have been waiting.
+        RfDelay(BlockNumber),
+    }
+
+    impl SpreadAllocate for Status {
+        #[inline]
+        fn allocate_spread(ptr: &mut KeyPtr) -> Self {
+            ptr.advance_by(<BlockNumber>::FOOTPRINT * 2);
+            Self::NotStarted
+        }
+    }
+    /// Struct for storing auction data.
+    #[derive(
+        Debug,
+        PartialEq,
+        scale::Encode,
+        scale::Decode,
+        Clone,
+        SpreadLayout,
+        PackedLayout,
+        SpreadAllocate,
+    )]
+    #[cfg_attr(feature = "std", derive(scale_info::TypeInfo, StorageLayout))]
+    pub struct Auction {
+        /// Branded name of the auction event
+        name: String,
+        /// Some hash identifiyng the auction subject
+        subject: Hash,
+        /// Structure storing the bids being made
+        bids: Bids,
+        /// Auction terms encoded as:
+        /// [start_block, opening_period, closing_period]
+        terms: [BlockNumber; 3],
+        /// Auction status
+        status: Status,
+        /// Candle auction can have no winner.
+        /// If auction is finalized, that means that the winner is determined.
+        finalized: bool,
+    }
+
+    impl Default for Auction {
+        fn default() -> Auction {
+            Auction {
+                name: String::default(),
+                subject: Hash::default(),
+                bids: Bids::default(),
+                terms: <[BlockNumber; 3]>::default(),
+                status: Status::OpeningPeriod,
+                finalized: false,
+            }
+        }
+    }
+
+    /// Way to fail a contract execution.
+    #[derive(scale::Encode, scale::Decode, Debug, PartialEq)]
+    #[cfg_attr(feature = "std", derive(scale_info::TypeInfo))]
+    pub enum Failure {
+        Revert(String),
+        Panic,
+    }
+
+    /// Storage of the contract.
+    #[ink(storage)]
+    #[derive(Default, SpreadAllocate)]
+    pub struct Mother {
+        auction: Auction,
+        balances: Mapping<AccountId, Balance>,
+    }
+
+    impl Mother {
+        #[ink(constructor)]
+        pub fn new(auction: Auction) -> Self {
+            initialize_contract(|c: &mut Self| {
+                c.balances = <Mapping<AccountId, Balance>>::default();
+                c.auction = auction;
+            })
+        }
+
+        #[ink(constructor)]
+        pub fn default() -> Self {
+            initialize_contract(|c: &mut Self| {
+                c.balances = <Mapping<AccountId, Balance>>::default();
+                c.auction = Auction::default();
+            })
```

Comment:
```suggestion
            Self::new(Auction::default())
```

Reply:
Why? Shouldn't we initialize the Mapping storage field, as our documentation [states](https://paritytech.github.io/ink-docs/datastructures/mapping)?

Reply:
right, gotcha